osu.Game/Overlays/UserProfileOverlay.cs

```diff
@@ -143,24 +170,27 @@ public void ShowUser(IUser user, IRulesetInfo? ruleset = null)
 
             sectionsContainer.ScrollToTop();
 
+            if (!API.IsLoggedIn)
+                return;
+
             userReq = user.OnlineID > 1 ? new GetUserRequest(user.OnlineID, ruleset) : new GetUserRequest(user.Username, ruleset);
-            userReq.Success += u => userLoadComplete(u, ruleset);
```

Comment:
Not sold by this change. The ruleset was being passed to the callback here intentionally, to isolate it from the ambient state of the user profile overlay. To illustrate, see following scenario:

- user profile overlay initiates request for profile of user A in osu! ruleset
- user profile overlay initiates request for profile of user B in mania ruleset
- request for profile of user A completes here, but the user profile overlay's `ruleset` field is now mania and not osu!, which is incorrect.

Maybe this won't ever cause an actual issue, but I have seen this go wrong enough times to be uncomfortable when seeing another instance of such a pattern.

Reply:
Have reverted this back to passing the ruleset into the callback - definitely feels more correct here